sql/core/src/test/scala/org/apache/spark/sql/connector/V2CommandsCaseSensitivitySuite.scala

```diff
@@ -316,17 +316,34 @@ class V2CommandsCaseSensitivitySuite extends SharedSparkSession with AnalysisTes
     }
   }
 
+  test("SPARK-36449: Replacing columns with duplicate name should not be allowed") {
+    alterTableTest(
+      () => ReplaceColumns(
```

Comment:
need to create a new `ReplaceColumns`. Otherwise, `analyzed` will be set to true after the first iteration.